components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java

```diff
@@ -140,6 +140,37 @@ public Tenant getTenant(String tenantUniqueID) throws TenantMgtException {
         return tenant;
     }
 
+    public Tenant getTenantByDomain(String domain) throws TenantMgtException {
+
+        TenantManager tenantManager = TenantMgtServiceComponent.getTenantManager();
+        Tenant tenant;
+        try {
+            int tenantID = tenantManager.getTenantId(domain);
+            tenant = (Tenant) tenantManager.getTenant(tenantID);
+            if (tenant == null) {
+                throw new TenantManagementClientException(ERROR_CODE_DOMAIN_NOT_FOUND.getCode(),
+                        String.format(ERROR_CODE_DOMAIN_NOT_FOUND.getMessage(), domain));
+            }
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            throw new TenantManagementServerException("Error while getting the tenant.", e);
```

Comment:
Isn't it good to have the tenant ID printed here ? Or domain so that we have an idea for which tenant the issue was caused ?

Reply:
+1 will add.

Reply:
Addressed.